### PR TITLE
Resimulation producer

### PR DESCRIPTION
**Pull request by EinarElen**
I am updating _ldmx-sw_, here are the details.

### What are the issues that this addresses?
This resolves https://github.com/LDMX-Software/SimCore/issues/49 by adding a new producer that re-simulates events based on the seeds stored in the input file. 

## Check List
- [x] I successfully compiled _ldmx-sw_ with my developments

- [x] I ran my developments and the following shows that they are successful.

I've added an analyzer in the DQM module that compares the sim hits and particles between two passes and tested it with an ecalPN configuration. 

- [x] I attached any sub-module related changes to this PR.

### Related Sub-Module PRs
If you have developments in a submodule that you need to merge in along with these developments, please link those PRs here, otherwise just delete this section.
_Note_: You do not need to make a Sub-Module PR if your submodule developments are already on that repo's default branch.

- https://github.com/LDMX-Software/SimCore/pull/95

### CODE REVIEW

**Comment by tomeichlersmith** [comment]
@EinarElen and I agreed to "hijack" this PR to do some code cleanup and reformatting within SimCore. While the newly introduced ReSimulator is not being tested by these PR Validations - it is helpful to make sure that the broad formatting/refactoring did not change any of the physics. I am not expecting it to since I trust clang to be helpful in this respect, but it is good to double check.

**Comment by tomeichlersmith** [comment]
I am getting a python-config error from the DQM module when trying to rerun the validation:
```
     File "/home/runner/work/ldmx-sw/ldmx-sw/.github/validation_samples/ecal_pn/config.py", line 51, in <module>
      from LDMX.DQM import dqm
    File "/home/runner/work/ldmx-sw/ldmx-sw/install/python/LDMX/DQM/dqm.py", line 23
      DQM/python/dqm.py 
                        ^
  IndentationError: unindent does not match any outer indentation level
```
Maybe there is an issue with some of the updates you made when writing the DQM resim verifier? I'll leave this to your tomorrow (my tonight) and come back at it tomorrow (mine).

**Comment by EinarElen** [comment]
> I am getting a python-config error from the DQM module when trying to rerun the validation:
> 
> ```
>      File "/home/runner/work/ldmx-sw/ldmx-sw/.github/validation_samples/ecal_pn/config.py", line 51, in <module>
>       from LDMX.DQM import dqm
>     File "/home/runner/work/ldmx-sw/ldmx-sw/install/python/LDMX/DQM/dqm.py", line 23
>       DQM/python/dqm.py 
>                         ^
>   IndentationError: unindent does not match any outer indentation level
> ```
> 
> Maybe there is an issue with some of the updates you made when writing the DQM resim verifier? I'll leave this to your tomorrow (my tonight) and come back at it tomorrow (mine).

Hmm, I would more suspect me messing up a rebase here. I'll have a look.

File: DQM/python/dqm.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ class HCalGeometryVerifier(ldmxcfg.Analyzer) :
         p.sequence.append( dqm.HcalGeometryVerifier() )
 
     """
-
+ DQM/python/dqm.py
```
Comment (EinarElen) [inline DQM/python/dqm.py]: Yeah that comes from a miss in a merge...

**Comment by tomeichlersmith** [comment]
The `workflow_dispatch:` will only be registered if the workflow **on trunk** has it.

**Comment by tomeichlersmith** [comment]
I used my god-like power to add `workflow_dispatch:` to the workflow on trunk and then put it back here. Then I was able to manually re-trigger the PR Validation run: https://github.com/LDMX-Software/ldmx-sw/actions/runs/5823063957

Unfortunately, it appears like GitHub is not registering that run as associated with this PR which is unfortunate.

**Comment by EinarElen** [comment]
Looks like the validation passed :)

**Review by tomeichlersmith — APPROVED** [review]
looks good to me - a bit worried about conflicts coming from me editing the pr_validation workflow on trunk and this branch but that will be easy for me to patch mid-merge.

**Comment by tomeichlersmith** [comment]
(I ended up just doing a non-rebase merge locally
```
git fetch --all
git switch trunk
git pull
git switch SimCore49-ReSim
git pull
git merge trunk
git switch trunk
git merge SimCore49-ReSim
git push
```
)

**Pull request merged**